# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/DCLayerTree.h
# Commit: f3b490c076d6
# Full Hash: f3b490c076d623d6ef3bcfd297a0522d9727477c
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-01-23 21:42:24
# Regressor Bug: 1602643
# File Overlap Count: 1
# Description:
#   Bug 1602643 - Disable WebRender compositor dinamically for async screenshot r=gw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D59658
# ==============================================================================

diff -r 0c5fb3a8a94d -r f3b490c076d6 gfx/webrender_bindings/DCLayerTree.h
--- a/gfx/webrender_bindings/DCLayerTree.h	Thu Jan 23 15:35:45 2020 +0200
+++ b/gfx/webrender_bindings/DCLayerTree.h	Thu Jan 23 04:37:28 2020 +0000
@@ -60,7 +60,9 @@
 
   void SetDefaultSwapChain(IDXGISwapChain1* aSwapChain);
   void MaybeUpdateDebug();
+  void MaybeCommit();
   void WaitForCommitCompletion();
+  void DisableNativeCompositor();
 
   // Interface for wr::Compositor
   void CompositorBeginFrame();
@@ -96,6 +98,7 @@
       wr::DeviceIntRect aDirtyRect, wr::DeviceIntPoint* aOffset,
       RefPtr<IDCompositionSurface> aCompositionSurface,
       wr::DeviceIntPoint aSurfaceOffset);
+  void ReleaseNativeCompositorResources();
 
   RefPtr<gl::GLContext> mGL;
   EGLConfig mEGLConfig;
@@ -147,6 +150,8 @@
   // TODO(gw): Might be faster as a hashmap? The length is typically much less
   // than 10.
   std::vector<CachedFrameBuffer> mFrameBuffers;
+
+  bool mPendingCommit;
 };
 
 /**